File: customer/frontend/lib/core/pricing/domain/ride_category.dart
enum RideCategoryType {
  bike,
  auto,
  miniSedan,
}

class RideCategory {
  final String id;
  final RideCategoryType code;
  final String displayName;
  final String description;
  final int capacity;

  const RideCategory({
    required this.id,
    required this.code,
    required this.displayName,
    required this.description,
    required this.capacity,
  });

  factory RideCategory.fromJson(Map<String, dynamic> json) {
    final codeStr = json['code'] as String;
    RideCategoryType code = RideCategoryType.miniSedan;
    if (codeStr == 'BIKE') code = RideCategoryType.bike;
    if (codeStr == 'AUTO') code = RideCategoryType.auto;

    return RideCategory(
      id: json['id'] as String,
      code: code,
      displayName: json['displayName'] as String,
      description: json['description'] as String,
      capacity: json['capacity'] as int,
    );
  }
}

File: customer/backend/src/modules/pricing/domain/ride-category.ts
export enum RideCategoryType {
  BIKE = 'BIKE',
  AUTO = 'AUTO',
  MINI_SEDAN = 'MINI_SEDAN',
}

export interface RideCategory {
  id: string;
  code: RideCategoryType;
  displayName: string;
  description: string;
  capacity: number;
}

export const CATEGORIES: Record<RideCategoryType, RideCategory> = {
  [RideCategoryType.BIKE]: { id: 'cat_bike', code: RideCategoryType.BIKE, displayName: 'GoRush Bike', description: 'Beat the traffic', capacity: 1 },
  [RideCategoryType.AUTO]: { id: 'cat_auto', code: RideCategoryType.AUTO, displayName: 'GoRush Auto', description: 'No haggling, just riding', capacity: 3 },
  [RideCategoryType.MINI_SEDAN]: { id: 'cat_mini', code: RideCategoryType.MINI_SEDAN, displayName: 'GoRush Mini', description: 'Comfortable sedans', capacity: 4 },
};
